bgpd/bgp_network.c

```diff
@@ -565,6 +565,11 @@ static int bgp_update_address(struct interface *ifp, const union sockunion *dst,
 		return 1;
 
 	prefix2sockunion(sel, addr);
+
+	if (IN6_IS_ADDR_LINKLOCAL(&addr->sin6.sin6_addr)) {
```

Comment:
braces {} are not necessary for single statement blocks

Reply:
This sounds like a workaround (hack) here.

Reply:
This is not  hack. As per RFC 2553,
The sin6_scope_id field is a 32-bit integer that identifies a set of
   interfaces as appropriate for the scope of the address carried in the
   sin6_addr field.  For a link scope sin6_addr sin6_scope_id would be
   an interface index.  For a site scope sin6_addr, sin6_scope_id would
   be a site identifier.  The mapping of sin6_scope_id to an interface
   or set of interfaces is left to implementation and future
   specifications on the subject of site identifiers.

Reply:
Please do no resolve unfinished review comments

Reply:
@sudhanshukumar22 I understand, but what I wanted to say is, that this is not the best place for handling this, IMO. Can't this be handled in `prefix2sockunion()`?

Reply:
@sudhanshukumar22 can we fix CI warnings and I'll push this?

Reply:
@ton31337 I have removed the curly braces. Please review and approve the changes.